### PR TITLE
Animal sorting algorithm

### PR DESCRIPTION
**Pull request by tbush03**
I connected the map_rfid UI page to the database. 

Now when we hit simulate on the map_rfid page, it will add the simulated animals to the database and save them.

### CODE REVIEW

File: ExperimentDatabase.py [hunk 1]
```diff
@@ -6,38 +6,34 @@ def __init__(self, file=":memory:"):  #call with file name as argument or no arg
         self._c = self._conn.cursor()
         try:
             self._c.execute('''CREATE TABLE experiment (
-                                experiment_id INTEGER PRIMARY KEY,
```
Comment (aithaprasad) [inline ExperimentDatabase.py:9]: Why are we removing experiment_id?

File: ExperimentDatabase.py [hunk 2]
```diff
@@ -51,17 +47,30 @@ def setup_experiment(self, name, species, uses_rfid, num_animals, num_groups, ca
...
+        cages_per_group = (int(num_animals)/int(num_groups)) / int(cage_max)
+        
+        group = 1
+        cage = 1
+        for x in range(1, int(num_groups)+1):
+            for cage in range(int(cages_per_group)):
+                self._c.execute(''' INSERT INTO cages (group_id, num_animals, full) 
+                                    VALUES (?, ?, ?)''',
+                                    (x, 0, 0))
+                self._conn.commit()
```
Comment (aithaprasad) [inline ExperimentDatabase.py:70]: Committing to the Database for each run of for loop is not ideal. Let's discuss how to deal with this. !

File: ExperimentDatabase.py [hunk 3]
```diff
@@ -75,20 +84,53 @@ def setup_measurement_items(self, items):
...
+
+    def _get_next_group(self):
+        self._c.execute("SELECT group_id, num_animals FROM groups WHERE full=0")
+        info = self._c.fetchone()
+        group_id = info[0]
+        num_group_animals = info[1] + 1
+        
+        self._c.execute("SELECT num_animals, num_groups FROM experiment")
+        animals = self._c.fetchone()
+        max = animals[0] / animals[1]
```
Comment (aithaprasad) [inline ExperimentDatabase.py:125]: What happens if it's not a perfect integer? Or shall we use integer division ?

File: ExperimentDatabase.py [hunk 4]
```diff
@@ -51,17 +47,30 @@ def setup_experiment(self, name, species, uses_rfid, num_animals, num_groups, ca
...
-            self._c.execute(''' INSERT INTO groups (name, num_animals) 
-                                VALUES (?, ?)''',
-                                (group, animals_per_group))
+            self._c.execute(''' INSERT INTO groups (name, num_animals, full) 
+                                VALUES (?, ?, ?)''',
+                                (group, 0, 0))
             self._conn.commit()
 
+    def setup_cages(self, num_animals, num_groups, cage_max):
+        cages_per_group = (int(num_animals)/int(num_groups)) / int(cage_max)
```
Comment (aithaprasad) [inline ExperimentDatabase.py:61]: Shall we use integer division instead of float? Or do we need float division mandatorily?

**Review by aithaprasad — CHANGES_REQUESTED** [review]

File: ExperimentDatabase.py (hunk 1 above)
Comment (tbush03) [reply ExperimentDatabase.py:9]: We are not using experiment_id for anything. We are creating a different file for each experiment.

File: ExperimentDatabase.py (hunk 3 above)
Comment (tbush03) [reply ExperimentDatabase.py:125]: We should ensure that it is a perfect interger when they are setting up the experiment.

File: ExperimentDatabase.py (hunk 4 above)
Comment (tbush03) [reply ExperimentDatabase.py:61]: I will switch it to integer divison

File: ExperimentDatabase.py (hunk 2 above)
Comment (tbush03) [reply ExperimentDatabase.py:70]: I think we need to insert everytime because we need to create that many cages.

Comment (tbush03) [reply ExperimentDatabase.py:70]: I will be working on this for my next PR

**Review by aithaprasad — APPROVED** [review]

**Pull request merged**